src/lib.rs

```diff
@@ -45,10 +48,22 @@
 #[cfg(test)] #[macro_use] extern crate log;
 #[cfg(all(windows, test))] extern crate advapi32;
 
-use std::cmp::Ordering;
-use std::error::Error;
-use std::fmt;
-use std::ops::{Add, Sub};
+#[cfg(feature = "std")]
+mod std_impl;
+
+#[cfg(feature = "nightly")]
+#[macro_use]
+extern crate alloc;
+
+#[cfg(feature = "std")]
+extern crate core;
```

Comment:
`rustc` complains about duplicate crate definition if it's there unconditionally.